### ISSUE TITLE
Mapped items not displaying correctly

### ISSUE THREAD

**Issue by andyward** [xBimTeam/XbimWebUI#33]
While investigating #26 I discovered what looks like a problem in the transform handling for mapped items meaning some parts of the scene are not being rendered. This seem most obvious in things like Windows.

E.g.In the following OneWallTwoWindows model, note the left window pane is missing (easy to see on black background) as is the right outer-windowframe/ledge (although the inner frame - a BREP - is present)
![image](https://cloud.githubusercontent.com/assets/923699/9544445/b7c0ff80-4d78-11e5-9134-276ec38bcc5a.png)

Interestingly, in fiddling with the xviewer code to display isolated shapes (and their maps) the shape display individually, just not collectively.
![image](https://cloud.githubusercontent.com/assets/923699/9544590/f50cd3ea-4d79-11e5-93ab-ed33dc2704b5.png)

It's like it only manages to render the first map correctly.

I think you can also see this problem in the tutorial samples http://xbimteam.github.io/XbimWebUI/4_Building_seen_from_everywhere.live.html

Note the window frames are missing, and some of the furniture (e.g. chairs) are missing geometry.

![image](https://cloud.githubusercontent.com/assets/923699/9544698/c19e550a-4d7a-11e5-89ec-5e443b6a45ce.png)
![image](https://cloud.githubusercontent.com/assets/923699/9544710/e13a675a-4d7a-11e5-9c20-0235839ed713.png)

Looks like there may be some glitches in the WebUI after all @martin1cerny  :smile:

**Comment by martin1cerny** [comment]
Thanks @andyward, this is the one for me. I've realized this while ago but didn't get to investigate it so far. It is most probably some silly bug in xViewer or more likely in the shader. I'll have a look at it.

**Comment by andyward** [comment]
Great. I'd not rule out the wexbim Geometry though. I wasn't sure that the transforms looked right where there are multiple 'repetitions'

**Comment by martin1cerny** [comment]
Yes, I know. But last time we had a look at this with Steve it was all right in Xplorer with the same wexbim data. So I'll double check transformations in WeXplorer.

**Comment by martin1cerny** [comment]
Hi @andyward , 
I'm trying to find the bug but with no success so far. I've also tried to draw isolated shapes but it behaves the same as for the whole product. How did you isolate the individual shapes? I used [this](https://github.com/xBimTeam/XbimWebUI/blob/develop/Xbim.WeXplorer/Viewer/xbim-model-handle.debug.js#L141) function to draw isolated products and modified [condition](https://github.com/xBimTeam/XbimWebUI/blob/develop/Xbim.WeXplorer/Viewer/xbim-model-handle.debug.js#L145) to draw isolated shapes. Any hint would be much appreciated.

**Comment by martin1cerny** [comment]
Right, I went back in time and found the problem. I was trying to optimise binary reading by using views instead of slices but it seems like underlying memory of the ArrayBuffer gets released even if it is used in multiple TypedArrays which are a view of that memory. I reverted these few lines of code and it is all right now.
![download](https://cloud.githubusercontent.com/assets/773078/9652841/2ab937b4-521f-11e5-822a-8d682b20921d.png)
